File: TEST/DAY_30_1N_ALERTDIALOG/app/src/main/java/com/example/velic/day_30_1_alertdialog/MyDialogFragment7.java
package com.example.velic.day_30_1_alertdialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

public class MyDialogFragment7 extends DialogFragment {
    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setMessage(" миллион коту?")
                .setIcon(R.drawable.ic_pets_black_24dp)
                .setTitle("Важно! Максимальный перепост")
                .setPositiveButton("OK", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        ((MainActivity) getActivity()).okClicked();
                    }
                })
                .setNegativeButton("Отмена", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        ((MainActivity) getActivity()).cancelClicked();
                    }
                });

        return builder.create();
    }
}

File: TEST/DAY_30_1_ALERT_Dialog_kt/app/src/main/java/com/example/day_30_1_alert_dialog_k/MyDialogFragment7.kt
package com.example.day_30_1_alert_dialog_k

import android.app.AlertDialog
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.fragment.app.DialogFragment
// Передать данные в активность
class MyDialogFragment7 : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
        builder.setMessage(" миллион коту?")
                .setIcon(R.drawable.ic_pets_black_24dp)
                .setTitle("Важно! Максимальный перепост")
                .setPositiveButton("OK") { dialog, id -> (activity as MainActivity).okClicked() }
                .setNegativeButton("Отмена") { dialog, id -> (activity as MainActivity).cancelClicked() }

        return builder.create()
    }
}
